xdsl/dialects/dlti.py

```diff
@@ -0,0 +1,52 @@
+"""
+The Data Layout and Target Information (DLTI) dialect is intended to
+hold attributes and other components pertaining to descriptions of
+in-memory data layout and compilation targets.
+
+https://mlir.llvm.org/docs/Dialects/DLTIDialect/
+"""
+
+from __future__ import annotations
+
+from xdsl.dialects.builtin import StringAttr
+from xdsl.ir import Attribute, Dialect, ParametrizedAttribute, TypeAttribute
+from xdsl.irdl import irdl_attr_definition, param_def
+from xdsl.parser import AttrParser
+from xdsl.printer import Printer
+from xdsl.utils.exceptions import VerifyException
+
+
+@irdl_attr_definition
+class DataLayoutEntryAttr(ParametrizedAttribute):
+    """
+    An attribute to represent an entry of a data layout specification.
+    https://mlir.llvm.org/docs/Dialects/DLTIDialect/#datalayoutentryattr
+    """
+
+    name = "dlti.dl_entry"
+
+    key: Attribute = param_def()
+    value: Attribute = param_def()
+
+    def print_parameters(self, printer: Printer):
+        printer.print_string("<")
+        printer.print_attribute(self.key)
+        printer.print_string(", ")
+        printer.print_attribute(self.value)
+        printer.print_string(">")
+
+    @classmethod
+    def parse_parameters(cls, parser: AttrParser) -> tuple[Attribute, Attribute]:
+        parser.parse_punctuation("<")
+        key = parser.parse_attribute()
+        parser.parse_punctuation(",")
+        value = parser.parse_attribute()
+        parser.parse_punctuation(">")
+        return key, value
+
+    def verify(self) -> None:
+        if not isinstance(self.key, StringAttr | TypeAttribute):
```

Comment:
This isn't ideal, I can try to continue thinking about a better way to do `AnyOf` constraints.